First attempt at providing solutions directory and stubbed code for Kafka Streams (#10)
confluentinc/learn-kafka-courses: kafka-streams/README.md

for the Kafka Streams course.  To get started you'll need to sign up

> for a CCloud account.  Follow the steps below to get everything ready to go.

Reviewer: ```suggestion
for a Confluent Cloud account.  Follow the steps below to get everything ready to go.
```

It's recommended that you do these exercises while following along with the [Kafka Streams 101 course](https://developer.confluent.io/learn-kafka/kafka-streams/get-started/).
The link for each module of the course related to the exercise description will be provided.

##  Confluent Cloud
 
1. Go to [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree?utm_source=learnkafka). 
   If you don’t already have a Confluent Cloud account you can create one here. Use promotion code <TBC> for money off your cloud bill.

> 2. Once logged into your cloud account click on the `Add cloud environment button` on the right-hand side of the screen.  Name the environment something like "kafka_streams_course_env".  You can also choose to use the `default` environment already present.  Using a new environment gives you the ability to keep resources separate on your Confluent account.

Reviewer: ```suggestion
2. Once logged into your cloud account click on the `Add cloud environment button` on the right-hand side of the screen.  Name the environment something like `kafka_streams_course_env`.  You can also choose to use the `default` environment already present.  Using a new environment gives you the ability to keep resources separate on your Confluent account.
```

   ![Create an environment or use the default one](images/confluent_environment.png)

> 3. Click on the environment you just created or the `default` environment tile. Create a new cluster in Confluent by clicking on`Create cluster on my own`.  For the purposes of all the exercise modules you can use the `Basic` type. 

Reviewer: ```suggestion
3. Click on the environment you just created or the `default` environment tile. Create a new cluster by clicking on `Create cluster on my own`.  For the purposes of all the exercise modules you can use the `Basic` type. 
```

   Then click the Schema Registry tab and follow the prompts to set up a schema registry on the provider of your choice. Once that is complete, go back to your cluster.
   ![Set up Schema Registry](images/schema-registry-setup.png)
5. Next click on `Data Integration` in the menu on the left, then select `Clients`, then the `Java` tile.
   ![Select clients and then select java](images/clients_java_step.png)  
   
6. After step three above, you'll go to a page where you can create credentials for your cluster and Schema Registry.                               
   ![Create Kafka cluster and Schema Registry credentials](images/create_ccloud_properties.png)

   1. Click on `Create Kafka cluster API key`.
   2. Copy your key and secret, name the file, then click `Download and continue`. (Your credentials will populate into the configurations boilerplate.)
   3. Click on `Create Schema Registry API key`.
   4. Copy your key and secret, name the file, then click `Download and continue`. (Your credentials will populate into the configurations boilerplate.)
   5. Make sure `Show API keys` is selected, then `Copy` the configurations in the window.

>    6. Create a file named `ccloud.properties` in the `src/main/resources directory` of the cloned repo. Then paste the configurations into a file named `ccloud.properties`. Note that this file is ignored and should never get checked into GitHub.

Reviewer: ```suggestion
   6. Create a file named `ccloud.properties` in the `src/main/resources` directory of the cloned repo. Then paste the configurations into a file named `ccloud.properties`. Note that this file is ignored and should never get checked into GitHub.
```
